sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -0,0 +1,179 @@
+//===------- matrix-unified.hpp - SYCL matrix extension ----*- C++ -*------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+#include <sycl/ext/oneapi/matrix/matrix-tensorcores.hpp>
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+namespace matrix {
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols, use Use,
+          layout Layout, typename T2>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_fill(Group sg,
+                  joint_matrix<T, Use, NumRows, NumCols, Layout, Group> &res,
+                  const T2 &v) {
+  std::ignore = sg;
+#if defined(__SYCL_DEVICE_ONLY__)
+#if defined(__NVPTX__)
+  res.wi_marray = v;
+#endif // defined(__NVPTX__)
+#else
+  std::ignore = res;
+  std::ignore = v;
+  throw runtime_error(
+      "This version of the matrix extension is only currently supported on "
+      "Nvidia devices",
+      PI_ERROR_INVALID_DEVICE);
+#endif // defined(__SYCL_DEVICE_ONLY__)
+}
+
+template <typename Group, typename S, typename T, size_t NumRows,
+          size_t NumCols, use Use, access::address_space Space,
+          std::enable_if_t<std::is_same<S, std::remove_const_t<T>>::value,
+                           bool> = true>
+void joint_matrix_load(
+    Group sg,
+    joint_matrix<S, Use, NumRows, NumCols,
```

Comment:
use here should be use::accumulator, right?

Reply:
Yes you're right. I've made this change.